Library/Homebrew/formula.rb

```diff
@@ -690,10 +695,37 @@ def bottle_prefix
     prefix+".bottle"
   end
 
-  # The directory where the formula's installation logs will be written.
+  # The directory where the formula's installation or test logs will be written.
   # @private
-  def logs
-    HOMEBREW_LOGS+name
+  def logs(log_type = :main)
+    case log_type
+    when :main
```

Comment:
I'm not sure that works, because the per-step subdirectories are created conditionally inside the sandboxed processes only as needed, and I think that counts as a modification to the parent directory. I'll test it to be sure.

Would a different name help?

I think it's useful to be able to get the path to the top log directory for a given formula, e.g. for code that doesn't know or care what all steps might exist under it, like if you wanted to `cd` the user to it for manual examination, or just create a tarball of everything under it.